### PR TITLE
✨(chore): Null-safety (closes #5)

### LINKED ISSUE #5: Null safety migration
hi
First, I am so thankful for what you have done here.
I am just wondering is there any plane for migrating null-safety??
or any way to use this package with null-safety version of dart and flutter?

### PR DESCRIPTION
**Pull request by HugoHeneault**
@tuarrep Would you please review this null-safety migration? :)

Thanks!

### CODE REVIEW

**Comment by tuarrep** [comment]
I'll have a look in this next week. Thanks for your contribution.

File: lib/map_elevation.dart [hunk 1]
```diff
@@ -18,35 +18,40 @@ class Elevation extends StatefulWidget {
...
-  final ElevationGradientColors elevationGradientColors;
+  final ElevationGradientColors? elevationGradientColors;
 
   /// [WidgetBuilder] like Function to add child over the graph
-  final Function(BuildContext context, Size size) child;
+  final Function(BuildContext context, Size size)? child;
 
   Elevation(this.points,
-      {this.color, this.elevationGradientColors, this.child});
+      {required this.color, this.elevationGradientColors, this.child});
```
Comment (tuarrep) [inline lib/map_elevation.dart]: Worth adding this breaking change in CHANGELOG.md

File: lib/map_elevation.dart [hunk 2]
```diff
@@ -18,35 +18,40 @@ class Elevation extends StatefulWidget {
...
     return LayoutBuilder(builder: (BuildContext context, BoxConstraints bc) {
       Offset _lbPadding = Offset(35, 6);
       _ElevationPainter elevationPainter = _ElevationPainter(widget.points,
           paintColor: widget.color,
-          elevationGradientColors: widget.elevationGradientColors,
+          elevationGradientColors: widget.elevationGradientColors ??
+              ElevationGradientColors(
+                  gt10: Colors.green,
+                  gt20: Colors.orangeAccent,
+                  gt30: Colors.redAccent),
```
Comment (tuarrep) [inline lib/map_elevation.dart]: This parameter should not be defaulted to anything to allow disabling elevation gradients.

See:  
https://github.com/OwnWeb/map_elevation/blob/b37c46d34699efc394d1f5c11c3b0de7d9a58501/lib/map_elevation.dart#L152-L168

File: lib/map_elevation.dart [hunk 3]
```diff
@@ -102,35 +108,21 @@ class _ElevationState extends State<Elevation> {
...
-      this.elevationGradientColors}) {
-    _min = (points
-                    .map((point) {
-                      if (point.altitude == null) return 0.0;
-                      return point.altitude;
-                    })
-                    .toList()
-                    .reduce(min) /
-                100)
+      required this.elevationGradientColors}) {
```
Comment (tuarrep) [inline lib/map_elevation.dart]: Should be optional

**Review by tuarrep — CHANGES_REQUESTED** [review]

File: lib/map_elevation.dart (hunk 1 above)
Comment (HugoHeneault) [reply lib/map_elevation.dart]: Hmm there's no point making it required, so I'll default it to transparent ;)

Comment (HugoHeneault) [reply lib/map_elevation.dart]: Well it seems the painter isn't painting with transparent color. I'll mark it as a BC.

File: lib/map_elevation.dart (hunk 2 above)
Comment (HugoHeneault) [reply lib/map_elevation.dart]: Right, will do it

**Comment by HugoHeneault** [comment]
@tuarrep Thanks for your review. I made the requested changes

**Review by tuarrep — APPROVED** [review]
LGTM

**Pull request merged**